core/src/main/scala/endless/core/typeclass/protocol/Encoder.scala

```diff
@@ -1,5 +1,5 @@
 package endless.core.typeclass.protocol
 
-trait Encoder[-A] {
-  def encode(a: A): Array[Byte]
-}
+/** Function to encode a value of type `A` into binary array
+  */
+trait Encoder[-A] extends (A => Array[Byte])
```

Comment:
This change and the similar change in `Decoder` implies that we can’t create an instance of `Decoder[A] with Encoder[A]`, although that was possible before. Is this intended?